assets/details-disclosure.js

```diff
@@ -35,14 +35,15 @@ customElements.define('details-disclosure', DetailsDisclosure);
 class HeaderMenu extends DetailsDisclosure {
   constructor() {
     super();
-    this.header = document.querySelector('header');
+    this.header = document.querySelector('.header-wrapper');
   }
 
   onToggle() {
-    document.body.classList.toggle('overflow-hidden', this.querySelectorAll('.mega-menu[open]').length);
+    if (!this.header) return;
+    this.header.preventHide = !!this.querySelectorAll('.mega-menu[open]').length;
```

Comment:
Love it! The only trade off that this creates is that if buyer triggers the menu and don't close it then go explore down the page, the header will not appear like expected on scroll up. Did I get this right?

If so, I am less concerned because the menu was open on user behalf and they can close it by clicking/tapping a link or click outside the menu (dropdown or megamenu)